REVIEWING.md

```diff
@@ -0,0 +1,317 @@
+
+It is generally fine to increase constraints without it being a breaking change. For example, if we made the `title` property required, that would not be breaking.
+
+There are some edge-cases though, for example if `shape` was changed to just `'square'`, that would be a breaking change because consumers might be checking for `box.shape === 'rounded'`, which would then be breaking. It's typically a quite easy thing for consumers to fix though. More generally, type unions and discriminated unions are quite troublesome in output types, as both adding and removing types from them are considered breaking changes.
+
+Another way to think about the rules for evolving output types is that the new type must be assignable to the old type. In this case for example `_box: OldBox = {} as NewBox`. It's not a silver bullet though, because of edge-cases like the one mentioned above.
+
+#### I/O Types
+
+Some types are considered both input and output types. For example, consider the following example:
+
+```ts
+type Point = {
+  x: number;
+  y: number;
+};
+
+function trimCoords(point: Point): Point;
+```
+
+In this case `Point` is both an input and output type. This means that the only changes we can make to the type that aren't breaking are the intersection of allowed changes between input and output types. In practice this only allows for the addition of new optional properties. Because of this constraint it is generally best to avoid using I/O types, and keep the input separated from the output.
+
+There are some cases where I/O types favor either input or output when it comes to API stability. For example, all types used by Utility APIs are I/O types, but the stability of the output is a lot more important than the stability of the input. That is because it's a lot easier for the single producer of the input interface to adapt to changes compared to all consumers of the API that use it as an output type.
+
+#### Identifying the Contract Direction
+
+The only way to identify the contract direction of a type is to look at the context in which it's being used. In particular this can be tricky when looking at individual type aliases and interfaces, as you need to look at the rest of the package exports to see how the type is being used.
+
+One important rule is that the context considered for any type is limited to only the package in which the type is declared. Just because a type is imported in a different package and used as an input type does not make it an input type.
+
+The following rules can be used to identify the direction of a type alias or interface:
+
+- If the type is used in an input context, for example function parameter, then it's an input type.
+- If the type is used in an output context, for example function return type, then it's an output type.
+- If the type is referenced by another type, then it inherits the direction of that type, except if referenced through a function callback, in which case the direction is reversed.
```

Comment:
Very enlightening ! I never considered this concept of type direction reversal via callbacks